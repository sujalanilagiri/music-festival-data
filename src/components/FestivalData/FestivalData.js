import React from 'react';
export default function FestivalData (props) {  
   const DisplayMapObject = ({mapObject}) => {     
    const sortedMap = new Map([...mapObject].sort());
    return sortedMap ?[...sortedMap.keys()].map((element,key) =>{   
     return (
       <ul>
         <li key={key}>
             {element}
             <DisplayArray key={key} arraytoDisplay = {mapObject.get(element)} />
         </li>
        </ul>
     );
     }) :null;
   }

   const display = Object.values(props.data).map((d, key) => {
     
        return (
          <div key={key} >
            <li key={key}>
              {d.recordLabel
              }
            </li>
            <DisplayMapObject  mapObject={d.bands} />
          </div>
          );
        });

        const DisplayArray = ({arraytoDisplay}) => {
            return arraytoDisplay ?arraytoDisplay.map((element,key) =>{
           return(  <ul>
                 <li key={key}>
                     {element}
                 </li>
             </ul>
            )}):null;
       
           }
        
    
        return(
          <div>
            <ul>
              { props.data ?display:null }
            </ul>
          </div>
        );
}