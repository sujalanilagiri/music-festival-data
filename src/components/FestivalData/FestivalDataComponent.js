import React, { Component } from "react";
import { connect } from "react-redux";
import FestivalData from "./FestivalData";
import { getFestivalData } from "./modules/FestivalDataActions";

const dataReceived = data => {
  return data
    ? data.reduce((acc, { name, bands }) => {
        bands.forEach(band => {
          var recordLabel =
            band["recordLabel"] && band["recordLabel"].length > 0
              ? band["recordLabel"]
              : undefined;

          var key = `${recordLabel}`;

          var bandName =
            band["name"] && band["name"].length > 0 ? band["name"] : undefined;

          if (!acc[key] && recordLabel && bandName)
            acc[key] = { bands: new Map(), recordLabel };

          if (recordLabel && bandName && name) {
            let festivals = [name];

            if (!acc[key].bands.has(bandName)) {
              acc[key].bands.set(bandName, festivals);
            } else {
              festivals = acc[key].bands.get(bandName);
              festivals.push(name);
              acc[key].bands.set(bandName, festivals);
            }
          }
        });
        return acc;
      }, {})
    : null;
};
const sortedData = data => {
  
  return data
    ? Object.entries(dataReceived(data))
        .sort()
        .reduce((o, [k, v]) => ((o[k] = v), o), {})
    : null;
};
class FestivalDataComponent extends Component {
  componentDidMount() {
    this.props.fetchData();
  }
  render() {
    if (this.props.response) var data = sortedData(this.props.response);
    return (
      <div>
        {this.props.response ? ( 
          <FestivalData key="festivalData" data={data} />
        ) : (
          "No Data received from Api to Display"
        )}
      </div>
    );
  }
}

const mapStateToProps = state => {
  return {
    response: state.festivalData.response
      ? state.festivalData.response.data
      : null
  };
};
const matchDispatchToProps = dispatch => ({
  fetchData: () => dispatch(getFestivalData())
});
export default connect(
  mapStateToProps,
  matchDispatchToProps
)(FestivalDataComponent);
