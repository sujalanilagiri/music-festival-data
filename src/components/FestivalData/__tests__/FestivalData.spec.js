import { shallow } from 'enzyme';
import * as React from 'react';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import  FestivalData  from '../FestivalData';

configure({adapter:new Adapter()});

let data =[
    {
      "name": "LOL-palooza",
      "bands": [
        {
          "name": "Jill Black",
          "recordLabel": "Fourth Woman Records"
        },
        {
          "name": "Werewolf Weekday",
          "recordLabel": "XS Recordings"
        },
        {
          "name": "Frank Jupiter",
          "recordLabel": "Pacific Records"
        },
        {
          "name": "Winter Primates",
          "recordLabel": ""
        }
      ]
    }
]

describe('<FestivalData />', () => {
  it('renders a button and a bash count', () => {
     let component = shallow(<FestivalData data={data}/>);
     expect(component).toMatchSnapshot();
  });
})