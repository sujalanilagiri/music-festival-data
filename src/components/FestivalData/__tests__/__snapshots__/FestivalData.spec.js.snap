// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FestivalData /> renders a button and a bash count 1`] = `ShallowWrapper {}`;
